tests/python/pants_test/engine/util.py

```diff
@@ -100,6 +101,7 @@ def create_scheduler(rules, validate=True, native=None):
     native,
     FileSystemProjectTree(os.getcwd()),
     './.pants.d',
+    safe_mkdtemp(),
```

Comment:
See my comment on the other review.

Reply:
I think isolation by default is good, because:
1. Isolation is generally good. I think @dotordogh or I ran into issues where https://github.com/pantsbuild/pants/blob/db713e4ed45633474b102a8e45483b55f277f1d2/tests/python/pants_test/engine/test_fs.py#L406-L423 was either failing when it should pass, or passing when it should fail, before the `prime_store_with_roland_digest` was added, because it relied on the store already happening to have the digest in question in it.
2. The speedup of re-using an existing store should be pretty small.

(The "force isolation" function is handy because the default in this PR is actually to share a store across an test class)

Happy to re-visit if you are convinced otherwise :)